JavaScript/2-doubly.js

```diff
@@ -28,17 +28,134 @@ LinkedList.prototype.pop = function() {
   }
 };
 
+LinkedList.prototype.shift = function(data) {
+  const node = new Node(this, data);
+  node.next = this.first;
+  if (this.length === 0) this.last = node;
+  else this.first.prev = node;
+  this.first = node;
+  this.length++;
+  return node;
+};
+
+LinkedList.prototype.unshift = function() {
+  if (this.length > 0) {
+    const node = this.first;
+    this.first = node.next;
+    node.list = null;
+    node.prev = null;
+    node.next = null;
+    this.length--;
+    return node.data;
+  }
+};
+
+LinkedList.prototype.findFirst = function(name){
+  return this.first.search(name);
+}
+
+LinkedList.prototype.insert = function (data, n) {
+  if (n <= 0){
+    return this.shift(data);
+  }else if (n >= this.length) {
+    return this.push(data);
+  }
+  else {
+    const node = new Node(this, data);
+    const insertAfterNode = this.first.searchPlace(n);
+    node.next = insertAfterNode.next;
+    node.prev = insertAfterNode;
+    insertAfterNode.next = node;
+    node.next.prev = node;
+    this.length++;
+    return node;
+  }
+};
+
+LinkedList.prototype.erase = function(n){
+  if (n <= 0){
+    return this.unshift(data);
+  }else if (n >= this.length) {
+    return this.pop(data);
+  }
+  else {
+    const eraseAfterNode = this.first.searchPlace(n);
+    const node = eraseAfterNode.next;
+    eraseAfterNode.next = node.next;
+    node.next.prev = eraseAfterNode;
+    node.list = null;
+    node.prev = null;
+    node.next = null;
+    this.length--;
+    return node.data;
+  }
+}
+
+LinkedList.prototype.append = function(data, n){
+  for (var i = 0; i < data.length; i++) {
+    this.insert(data[i], n + i);
+  }
+}
+
+LinkedList.prototype.findAll = function(name){
+  return list.first.searchAll(name);
+}
+
+LinkedList.prototype.find = function(name, func){
```

Comment:
LinkedList.find should be Array.find compatible
Ref: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/find